dev/src/Domain/Contract.java

```diff
@@ -28,8 +28,29 @@ public static Contract getContractFromIO(IO io, Supplier contractSupplier) {
         return new Contract(wholesaleThreshold, contractSupplier, products);
     }
 
+    public boolean hasProduct(String productName) {
+        return products.stream().anyMatch(product -> product.name().equals(productName));
+    }
+
+    public double getPriceOfProduct(ProductInOrder productInOrder) {
```

Comment:
It's weird to me that Contract function gets productInOrder.
Why do you check the min value? We're not assuming that product can be only once in contract?